server/services/store/sqlstore/board_insights.go

```diff
@@ -0,0 +1,134 @@
+package sqlstore
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/mattermost/focalboard/server/model"
+
+	sq "github.com/Masterminds/squirrel"
+
+	"github.com/mattermost/mattermost-server/v6/shared/mlog"
+)
+
+func (s *SQLStore) getTeamBoardsInsights(db sq.BaseRunner, teamID string, duration string) ([]*model.BoardInsight, error) {
+	/**
+	Some squirrel issues to note here are
+	1. https://github.com/Masterminds/squirrel/issues/285 - since we're using 1+ sub queries. When placeholders are counted for second query, the placeholder names are repeated.
+		This is the reason to not use conditional operators in Where clauses which would eventually parametrize the variables.
+	*/
+
+	boardsHistoryQuery := s.getQueryBuilder(db).Select("boards.id", "boards.title", "boards.icon",
+		"count(boards_history.id) as count", "boards_history.modified_by", "boards.created_by").
+		From(s.tablePrefix + "boards_history as boards_history").
+		Join(s.tablePrefix + "boards as boards on boards_history.id = boards.id").
+		Where(fmt.Sprintf("boards_history.insert_at > %s and boards.team_id = '%s' and boards_history.modified_by != 'system' and boards.delete_at = 0",
+			s.durationSelector(duration), teamID)).
+		GroupBy("boards_history.id, boards.id, boards_history.modified_by")
+	blocksHistoryQuery := s.getQueryBuilder(db).Select("boards.id", "boards.title", "boards.icon",
+		"count(blocks_history.id) as count", "blocks_history.modified_by", "boards.created_by").
+		From(s.tablePrefix + "blocks_history as blocks_history").
```

Comment:
This would be too much data to fetch into memory at once. It won't impact new users but the more you'll use Boards, the more this query will suffer. Can we perform this operation in batch or think of any other way to make it more performant?

Where exactly and how frequently will this API be called BTW?

Reply:
@harshilsharma63 Yeah, I'm sure we can think of lots of ways to make it more performant; the sky's the limit. I've been testing this against one of our loadtest instances which has a very small dataset. Before this feature can be enabled for customers we'll need to add loadtest data and tweak the performance to an acceptable level.